rethink.js

```diff
@@ -599,14 +599,14 @@ RethinkDB.prototype._all = function _all(model, filter, options, callback) {
             return callback(error, null);
         }
 
-        _keys = _this._models[model].properties;
-        _model = _this._models[model].model;
-
         cursor.toArray(function (err, data) {
             if (err) {
                 return callback(err);
             }
 
+            _keys = _this._models[model].properties;
+            _model = _this._models[model].model;
+
```

Comment:
@hpstuff thanks for the contribution.
I had a look at the code, and seems to be present another issue, _keys and _model variables are not initialised with "var" statement.
It's a previous bug not introduced by you.  Could you fix it please?